nft_ingester/src/api/api_impl.rs

```diff
@@ -382,6 +386,35 @@ where
         Ok(json!(res))
     }
 
+    pub async fn get_nft_editions(&self, payload: GetNftEditions) -> Result<Value, DasApiError> {
+        let label = "get_nft_editions";
+        self.metrics.inc_requests(label);
+        let latency_timer = Instant::now();
+
+        let pagination = Pagination {
+            limit: payload.limit.unwrap_or(DEFAULT_LIMIT as u32).into(),
+            page: payload.page,
+            before: payload.before,
+            after: payload.after,
+            cursor: payload.cursor,
+        };
+        Self::validate_basic_pagination(&pagination, self.max_page_limit)?;
+        let mint_address = Pubkey::from_str(&payload.mint)
+            .map_err(|_| DasApiError::Validation("Invalid mint address".to_string()))?;
+
+        let master_asset_editions_info = self
+            .rocks_db
+            .get_master_edition_child_assets_info(
+                mint_address,
+                PaginationQuery::build(pagination.clone()).unwrap(),
```

Comment:
is this unwrap necessary? seems like we can build the builder with proper error handling.

Reply:
Without unwrap we have Result< PaginationQuery > but function expected PaginationQuery

Reply:
You can use `?` on it before the function call and handle the error instead of panicking.